### ISSUE TITLE
Compiler Defaults to P2 Syntax

### ISSUE THREAD

**Issue by JonMcPhalen** [parallaxinc/Propeller-Tool#20]
I have a P1 connected and used F7 to identify. When I attempt to compile the code the IDE flags a syntax error: it is expecting "(" after method names (P2 syntax). There is no P2 connected, nor any tabs open with P2 code.

**Comment by PropGit** [comment]
Is the code saved yet (ie: is it still nameless)?  Without the source directives (which don't exist yet), it doesn't have any reasonable way to detect the intended target other than the file extension.  That's a case where we really need the directives.

I think it's using whatever target was left over from a previous source operation.

**Comment by PropGit** [comment]
I verified that it does the same to me but haven't figured out why it tries compiling for P2; however, once saved with a .spin extension, it compiles properly.

This will be improved when the directives are added.  Keeping issue open to ensure it's resolved.

**Comment by JonMcPhalen** [comment]
Saving with .spin extension worked.

**Comment by PropGit** [comment]
I've determined why it's defaulting to compiling for P2.  When the source type is unknown as of yet (ie: file not yet saved so there's no extension), the compiler code defaults to P2.  This was unintentional.  I've now made it default to P1 unless it's specifically known to be P2 code (a .spin2 extension for now and/or (later) a Propeller 2 directive exists in the source).

This change will be in v2.1+ Alpha.

The result is that you now won't have this problem as detailed at the start of this issue; however, if you use a P2 and unsaved code (intended for P2), it will complain when non-P1 code is detected - this is the opposite problem from what you described.  This is just for your information.  The final intent is to implement a Propeller directive and add in a prompt telling the user the source needs a Propeller directive (when the source type can't be inferred by the file extension) upon compilation.  The directive will likely be in v2.2 Alpha or v2.3 Alpha.
